cordapp-example/workflows-kotlin/src/main/kotlin/com/example/flow/PaybackFlow.kt

```diff
@@ -169,27 +155,23 @@ object PaybackFlow {
 
             val signTransactionFlow = object : SignTransactionFlow(borrowerPartySession) {
                 override fun checkTransaction(stx: SignedTransaction) = requireThat {
-                    val allInputStates = stx.tx.inputs.getStxStates(serviceHub)
+                    val allInputStates = stx.tx.inputs.map {
+                        serviceHub.toStateAndRef<ContractState>(it).state.data
+                    }
 
                     val inIOUState = allInputStates.filterByState<IOUState>().single()
                     val inCashStates = allInputStates.filterByState<CashState>()
 
                     val outCashStates = stx.tx.outputsOfType<CashState>()
-                    val lenderPaybackCash = outCashStates
-                            .stream().filter { it.owner.equals(inIOUState.lender) }
-                            .toList()
-
-                    val inputCashSum = inCashStates.sumByLongSecure { it.value }
-                    val lenderPaybackSum = lenderPaybackCash.sumByLongSecure { it.value }
 
-                    "Expect exact payback value. $lenderPaybackSum != ${inIOUState.value}" using (lenderPaybackSum == inIOUState.value.toLong())
-
-                    val borrowerChange = outCashStates
-                            .stream().filter { it.owner.equals(inIOUState.borrower) }
+                    val lenderPaybackCash = outCashStates
+                            .filter { it.owner.equals(inIOUState.lender) }
                             .toList()
                             .sumByLongSecure { it.value }
-                    "Expect change for borrower." using (inputCashSum - lenderPaybackSum == borrowerChange)
 
+                    "Expect exact payback value. $lenderPaybackCash != ${inIOUState.value}" using (lenderPaybackCash == inIOUState.value.toLong())
+
+                    "All input money must be owned by borrower." using inCashStates.all { it.owner == inIOUState.borrower }
```

Comment:
This is restrictive. I am ok with it. If you wanted to allow cash states coming from multiple sources, except the lender, then you could use `!= lender` instead.

Reply:
hmmm... how can I allow borrower to do payback with not his money? Well... ok maybe I could if I will ask for signatures of all owners on input cash, right? but let's move forward ;)